### PR TITLE
Update SA1600 to also handle records

### LINKED ISSUE #3723: SA1600 doesn't seem to work for record types
SA1600 doesn't seem to report missing documentation for record types. Is this a bug or by design?

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3723

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs [hunk 1]
```diff
@@ -24,7 +25,8 @@ namespace StyleCop.Analyzers.DocumentationRules
     ///
     /// <para>A violation of this rule occurs if an element is completely missing a documentation header, or if the
     /// header is empty. In C# the following types of elements can have documentation headers: classes, constructors,
-    /// delegates, enums, events, finalizers, indexers, interfaces, methods, properties, and structs.</para>
+    /// delegates, enums, events, finalizers, indexers, interfaces, methods, properties, records, record classes,
+    /// record structs and structs.</para>
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs:29]:
```suggestion
    /// delegates, enums, events, finalizers, indexers, interfaces, methods, records, properties, and structs.</para>
```

File: documentation/SA1600.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ A C# code element is missing a documentation header.
 
 C# syntax provides a mechanism for inserting documentation for classes and elements directly into the code, through the use of Xml documentation headers. For an introduction to these headers and a description of the header syntax, see the following article: [https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments).
 
-A violation of this rule occurs if an element is completely missing a documentation header, or if the header is empty. In C# the following types of elements can have documentation headers: classes, constructors, delegates, enums, events, finalizers, indexers, interfaces, methods, properties, and structs.
+A violation of this rule occurs if an element is completely missing a documentation header, or if the header is empty. In C# the following types of elements can have documentation headers: classes, constructors, delegates, enums, events, finalizers, indexers, interfaces, methods, properties, records, record classes, record structs and structs.
```
Comment (sharwell) [inline documentation/SA1600.md:26]:
```suggestion
A violation of this rule occurs if an element is completely missing a documentation header, or if the header is empty. In C# the following types of elements can have documentation headers: classes, constructors, delegates, enums, events, finalizers, indexers, interfaces, methods, properties, records, and structs.
```

**Comment by sharwell** [comment]
The implementation change is easy enough. Two small tweaks to the documentation (simplify to just 'records', and add the missing Oxford comma).

**Comment by bjornhellander** [comment]
Applied suggestions

**Review by sharwell — APPROVED** [review]

**Pull request merged**